server/src/main/java/org/candlepin/resteasy/filter/VerifyAuthorizationFilter.java

```diff
@@ -111,39 +110,79 @@ public void runFilter(ContainerRequestContext requestContext) {
         }
     }
 
-    protected Map<Verify, Object> getArguments(HttpRequest request, Method method) {
-        ResteasyProviderFactory resourceFactory = ResteasyProviderFactory.getInstance();
-        InjectorFactory injectorFactory = resourceFactory.getInjectorFactory();
+    /**
+     * Fetches the arguments for parameters flagged with the Verify annotation from the request
+     * context. If the method does not have any Verify-annotated parameters, this method returns
+     * an empty map.
+     *
+     * @param requestContext
+     *  the context data for the request
+     *
+     * @param method
+     *  the method handling the request
+     *
+     * @return
+     *  a mapping of verify annotations to their respective arguments
+     */
+    protected Map<Verify, Object> getArguments(ContainerRequestContext requestContext, Method method) {
+
+        // LinkedHashMap preserves insertion order
+        Map<Verify, Object> argMap = new LinkedHashMap<>();
 
-        ResourceLocator locator = locatorMap.get(method);
+        Annotation[][] annotations = annotationLocator.getParameterAnnotations(method);
 
-        if (null == locator) {
-            throw new IseException("Method " + method.getName() + " not registered as RESTful.");
-        }
+        Map<String, List<String>> pathParams = null;
+        Map<String, List<String>> queryParams = null;
+        Map<String, List<String>> headers = null;
 
-        MethodInjector methodInjector = injectorFactory.createMethodInjector(locator, resourceFactory);
-        HttpResponse response = ResteasyProviderFactory.popContextData(HttpResponse.class);
-        Object[] args = methodInjector.injectArguments(request, response);
+        for (int i = 0; i < annotations.length; ++i) {
```

Comment:
This could be enhanced for

Reply:
I suppose, but this was something that was pre-existing that would get (immeasurably) slower by a change, so I opted to just leave it as-is.